var/spack/repos/builtin/packages/spdk/package.py

```diff
@@ -0,0 +1,101 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import os
+
+from spack.package import *
+
+
+class Spdk(AutotoolsPackage):
+    """The Storage Performance Development Kit (SPDK) provides a set of tools
+    and libraries for writing high performance, scalable, user-mode storage
+    applications. It achieves high performance by moving all of the
+    necessary drivers into userspace and operating in a polled mode instead
+    of relying on interrupts, which avoids kernel context switches and
+    eliminates interrupt handling overhead.
+    """
+
+    homepage = "https://spdk.io"
+    url = "https://github.com/spdk/spdk/archive/v23.01.tar.gz"
+    git = "https://github.com/spdk/spdk"
+    maintainers("hyoklee")
+
+    version("master", branch="master", submodules=True)
+    version("23.01", tag="v23.01", submodules=True)
+    version("22.01.1", tag="v22.01.1", submodules=True)
+
+    variant("crypto", default=False, description="Build vbdev crypto module")
+    variant("fio", default=False, description="Build fio plugin")
+    variant("iscsi-initiator", default=False, description="Build with iscsi bdev module")
+    variant("ocf", default=False, description="Build OCF library and bdev module")
+    variant("pmdk", default=False, description="Build persistent memory bdev")
+    variant("rbd", default=False, description="Build Ceph RBD bdev module")
+    variant(
+        "rdma", default=False, description="Build RDMA transport for NVMf target and initiator"
+    )
+    variant("shared", default=False, description="Build spdk shared libraries")
+    variant("uring", default=False, description="Build I/O uring bdev")
+    variant(
+        "virtio", default=False, description="Build vhost initiator and virtio-pci bdev modules"
+    )
+    variant("vhost", default=False, description="Build vhost target")
+    variant("vtune", default=False, description="Profile I/O under Intel VTune Amplifier XE")
+
+    mods = (
+        "crypto",
+        "iscsi-initiator",
+        "ocf",
+        "pmdk",
+        "rbd",
+        "rdma",
+        "shared",
+        "uring",
+        "vhost",
+        "virtio",
+        "vtune",
+    )
+
+    depends_on("dpdk@22.11:")
```

Comment:
are you using an external dpdk in that case also ? where is it specified in the command line ?